### PR TITLE
Fix logging setup on Debian

### PR DESCRIPTION
**Pull request by ekohl**
This mitigates DSA-3701 / CVE-2016-1247 and matches the default setup on Debian.

The CVE mentioned Gentoo needs a similar treatment but I haven't looked into that yet.

https://www.debian.org/security/2016/dsa-3701
https://security-tracker.debian.org/tracker/CVE-2016-1247

### CODE REVIEW

**Comment by alexjfisher** [comment]
Does this not fix the issue on RedHat systems too? (even if the RPMs never had this bug, did this module introduce it for RedHat users)?

**Comment by ekohl** [comment]
I do not know. I just remembered this CVE when I saw this module was trying to change it.

File: spec/classes/nginx_spec.rb [hunk 1]
```diff
@@ -360,9 +360,9 @@
             it do
               is_expected.to contain_file('/var/log/nginx').with(
                 ensure: 'directory',
-                owner: 'www-data',
+                owner: 'root',
                 group: 'adm',
-                mode: '0750'
+                mode: '0755'
```
Comment (dhollinger) [inline spec/classes/nginx_spec.rb]: This test is failing on RHEL-based tests.

Comment (ekohl) [reply spec/classes/nginx_spec.rb]: Looks like that's a valid failure because I copy-pasted an incorrect default

**Pull request merged**